### PR TITLE
[READY] Renames existing skin colors, adds skin colors, updates spray tan code

### PR DESCRIPTION
**Pull request by OnlineGirlfriend**
## About The Pull Request

This renames existing skin colors while adding four new skin colors available for character preferences. The corresponding spray tan code has been updated to reflect this. This also adds the secret color "darkorange," a result of adding spray tan to the Cacao skin color. No skin colors were removed, and the albino and orange skin colors have been preserved.

Skin colors on existing saves will remain the same hex but convert to the new color name.

Colors will be listed in preferences from lightest to darkest.

Colors will appear in the following order:

Albino
Porcelain _(previously caucasian1)_
Ivory _(previously asian1)_
Sand _(previously caucasian2)_
Beige _(previously asian2)_
Peach _(previously caucasian3)_
Honey **(new)**
Sienna _(previously latino)_
Olive _(previously mediterranean)_
Mocha _(previously arab)_
Amber _(previously indian)_
Bronze **(new)**
Caramel _(previously african1)_
Hazelnut **(new)**
Espresso _(previously african2)_
Cacao **(new)**

Added colors:

Honey
Skin-Honey

Bronze
Bronze

Hazelnut
Skin-Hazelnut

Cacao
Skin-Cacao

Dark Orange (not available in character preferences)
DarkOrange

The full spectrum of skin colors available in character preferences per this PR:

spectrum-names

## Why It's Good For The Game

- The existing naming conventions were inconsistent. This creates a consistent naming standard across skin colors.
- It is now easier to quickly find your desired skin color, as the names are based on actual color.
- This looks better in-game and is more consistent with the other character preferences.
- The darkest skin color is now reactive to spray tan.

## Changelog
:cl:
add: new skin colors: Honey, Bronze, Hazelnut, Cacao
add: skin color: darkorange
tweak: skin color names
/:cl:

### CODE REVIEW

**Comment by Ty-the-Smonk** [comment]
Why coffee?

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 1]
```diff
@@ -335,28 +335,38 @@
...
+						N.skin_tone = "Olive"
+					if("Peach")
+						N.skin_tone = "Sienna"
+					if("Sand")
+						N.skin_tone = "Peach"
+					if("Porcelain")
+						N.skin_tone = "Sand"
+					if("Albino")
+						N.skin_tone = "Porcelain"
+					if("Cacao")
```
Comment (Tlaltecuhtli) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: N

**Comment by PKPenguin321** [comment]
some of these are good and some aren't. for example, what the fuck is sienna? i googled it and it's a car apparently.
image
if i didn't already know it was supposed to be "latino" i would have absolutely no clue what color it is supposed to be

**Comment by PKPenguin321** [comment]
also pics of the new colors please

**Comment by Akrilla** [comment]
finally

**Comment by WJohn** [comment]
meme pr is a meme pr

**Comment by OverlyGeneric** [comment]
@PKPenguin321 

Apparently, it looks like this but this looks more African than Latino.

image

**Comment by OnlineGirlfriend** [comment]
> @PKPenguin321
> 
> Apparently, it looks like this but this looks more African than Latino.
> 
> image

The idea is to group the skins by color instead of race, but also to re-purpose existing skin colors should people prefer to keep what they had.

I'm open to re-naming Sienna. However, players can just click on it to view it. The skin colors will be ordered from lightest to darkest to make selection easier.

**Comment by OnlineGirlfriend** [comment]
Here's another reference for sienna in regards to skin tone.
axsimen180100006

File: code/modules/reagents/chemistry/reagents/other_reagents.dm (hunk 1 above)
Comment (optimumtact) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm]: cut it out

**Comment by optimumtact** [comment]
you need to support the old tones because they're saved in preference files

File: code/modules/reagents/chemistry/reagents/other_reagents.dm (hunk 1 above)
Comment (Militaires) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm]: I

**Comment by OnlineGirlfriend** [comment]
> you need to support the old tones because they're saved in preference files

I did some testing - when you load the saved preferences, it just picks a new color at random. It doesn't break anything.

**Comment by PKPenguin321** [comment]
that's not what he's asking of you though

**Comment by optimumtact** [comment]
You can't just randomly change peoples skin tones, you have to convert from like to like dude.

**Comment by optimumtact** [comment]
Can you just add back the actual names instead, yours are silly

**Comment by nicbn** [comment]
Just make it show colored buttons for selection so that we won't have to name the skin tones

**Comment by OnlineGirlfriend** [comment]
Not stale, doing a push tonight.

**Comment by OnlineGirlfriend** [comment]
This commit isn't working the way that it should, I'm also encountering a glitch that is causing saved character names to be overwritten even with random name set to off. 

If anyone could look at this code and help me understand what I'm doing wrong here, that would be a great help.

**Comment by OnlineGirlfriend** [comment]
The required change has been made and this has been tested successfully.

**Comment by OnlineGirlfriend** [comment]
@optimumtact

The current names are inconsistent for expansion.

caucasian1, caucasian2 but arab, indian, mediterranean makes no sense.

I don’t feel like you have to like my color names to agree that the current ones aren’t good.

**Comment by PKPenguin321** [comment]
The current names are a lot clearer than these ones even if they're not perfect

**Comment by RaveRadbury** [comment]
The old names are dated as hell and make the game look bad. They aren't even extensible because the galaxy brain who made them decided to not put "1" at the end of several of the skin tones. 

This is a good change because it decouples a bunch of dated garbage from a needed system. It's sad y'all can't recognize that.

Welcome to the 90's, where there are 2 colors of African and only one color of Latino. What? Those are geographical and cultural nomenclatures unrelated to actual skin tones? Nah, totally legit.

**Comment by optimumtact** [comment]
lot to unpack there

**Comment by FantasticFwoosh** [comment]
> I couldn't tell from a glance without the actual character overlay what these would actually do or look like visualising them in my head. 

Ochre is a better definition of mocha/honey to hazelnut in existing etymology, but i don't have any support for this as it seems entirely over saturated with brown distinctless and classless palettes than the existing ones that give a kind of idea of who the player is meant to be which is important for character building. Seems very _PC_ motivated also to try and banish terms like 'Indian' or 'Caucasian' in broad strokes.

**Comment by fludd12** [comment]
Figured I'd toss my opinion on the pile, too. While the new names are somewhat confusing at first, given the fact that I'm pretty sure a large portion of our playerbase doesn't know the difference between, say, beige and peach, I appreciate the new name scheme over "dated, possibly racially insensitive names with numbers after them." The ordering them based on color lightness also makes the system a lot easier to make sense of.

While having something like an actual, though limited, color selection might be preferable, this is definitely a step in the right direction, and adds a fairly significant amount of character customization as well.

@PKPenguin321 @optimumtact I'm sorry to call you two out specifically, because I know you're not the only ones arguing the point, but you are some of the most visible and influential contributors here. I understand that your view is that the new system makes for slightly confusing name terminology, but personally, I end up clicking through every single skintone as-is when making a character, just because the difference between caucasian1 and caucasian3 is more than the difference between caucasian1 and asian1. Neither system is perfect, but this one at least orders them in a logical way, and can be more easily remembered when using it repeatedly.

**Comment by Militaires** [comment]
image
Oh dear... Yikes... you sweet summer child, who hurt you? It's called being a decent fucking human being, let that sink in.

**Comment by OnlineGirlfriend** [comment]
Seems very _PC_ motivated also to try and banish terms like 'Indian' or 'Caucasian' in broad strokes.

It’s actually because I loathe inconsistencies. “Caucasian” is a concept of race whereas “Asian” and “African” refer to a continent (there are Caucasian Africans, so idk), “Indian” refers to a country, “Mediterranean” refers to region with notably differing ethnic groups. None of these terms are offensive, they’re just nonsensical when lumped all together. Not to mention that any human of any ethnicity can have any skin color. So if people really want ethnicity role play then there should probably be a separate pref for this.

**Comment by OnlineGirlfriend** [comment]
Also, why would Latino be just a slightly darker pink than Caucasian? It doesn’t make sense. Again, this preserves the colors people were already using, just disconnects them from these names that don’t belie any information about what they are.

**Comment by RaveRadbury** [comment]
>the existing ones that give a kind of idea of who the player is meant to be which is important for character building. Seems very _PC_ motivated also to try and banish terms like 'Indian' or 'Caucasian' in broad strokes.

Yeah because in space everyone cares that you're African 2. It's not even Space-African2. Also who needs skin colors labelled after cultural groups in order to build your character? This is the kind of backwards thinking I'm talking about. The complexities between skintone and cultural group are far more interesting, given that a lot of cultures (including Latin culture) have biases towards light skin that affect class and discrimination. You can't even play a light or dark skinned Latino in our system. That lacks complexity. By this logic removing the connotations would actually add more freedom to character creation.

If this was a _PC_ motivated PR it would be something more like "Chat filters racial slurs". Skin tone charts aren't made by SJW's, they're made by cosmetologists who have to do make up work on different skin tones day in and day out and don't want to get hung up on grilling clients about their ethnicity (as its clearly pointless given the skin tone variation within ethnic groups). I get that many /tg/station community members haven't had to interact with this area of knowledge as the ones who /do/ use makeup likely only interact with /their own/ skin tone. But in a system where people should have the full character creator of skin tones available to them it should be separate from ethnic/racial/cultural connotation (as one skin tone could apply to multiple categories that way).

Dang @FantasticFwoosh you concerned about the "Caucasians" being "banished"? We're deliberately trying to avoid that kind of talk entirely and you're taking use straight into "You will not replace us" territory.

**Pull request closed without merging**